public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+            tmp.push(html`<${FutureScheduleCard}
+              time=${time}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+            />`);
+          } else {
+            let time =
+              "Now - " +
+              new Date(event.end_time).toLocaleTimeString().slice(0, -3);
+            tmp.push(html`<${LiveScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              time=${time}
+            />`);
+          }
+        }
+      });
+      if (tmp.length == 1) {
+        // No Interviews, Only Search
+        if (searchTerm.current == "") {
+          setSlots([html`<h2 class="text-center">Coming Soon...</h2>`]);
+        } else {
+          setSlots(
+            tmp.concat([
+              html`<br />
+                <h2 class="text-center">No Results</h2>`,
+            ])
+          );
+        }
+      } else {
+        setSlots(tmp);
+
+        if (youtubeTitle != "") {
+          setYoutubeVid(getLatestODVideo());
+
+          if (youtubeVid == defaultYouTube) {
+            setYoutubeTitle("On Demand Videos Available Soon!");
+          } else {
+            setYoutubeTitle("You Just Missed");
+          }
+        }
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!searched) {
+      updateSchedule(false);
+    } else if (searchTerm.current == "") {
+      // Saves generating loads of refreshes by searching
+      setTimeout(() => {
+        updateSchedule(true);
+      }, refreshTime);
+    }
+    setSearched(true);
+  });
+
+  var youtubeColumn = "";
+  if (isCINlive) {
+    youtubeColumn = html`
+      <div class="col">
+        <div
+          style="display: flex; position: -webkit-sticky;position: sticky;top: 33vh;"
+        >
+          <div>
+            <h3 class="text-center">${youtubeTitle}</h3>
+            <iframe
+              src="https://www.youtube.com/embed/${youtubeVid}"
+              width="600"
+              height="338"
+              style="border:none;overflow:hidden; max-width: 90%;"
+              scrolling="no"
+              frameborder="0"
+              allowTransparency="true"
+              allow="encrypted-media"
+              allowfullscreen="true"
+            ></iframe>
+            <br /><a
+              href="https://www.youtube.com/watch?v=${youtubeVid}"
+              class="cin-text-2"
+              >[External Link]</a
+            >
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  return html`
+    <div>
+      <h1 class="display-3 cin-text text-center">All Interviews</h1>
+      <div class="row">
+        <div class="col">${slots}</div>
+        ${youtubeColumn}
+      </div>
+    </div>
+  `;
+};
+
+const LiveArea = () => {
```

Comment:
You... You didn't just define a component _inside_ a component, did you? Please never do that again.

Reply:
No? LiveArea is sepeate from ScheduleArea

Reply:
oh yeah might be blind